modules/javafx.controls/src/main/java/com/sun/javafx/scene/control/behavior/TextFieldBehavior.java

```diff
@@ -69,12 +72,11 @@ public TextFieldBehavior(final TextField textField) {
             contextMenu.getStyleClass().add("text-input-context-menu");
         }
 
-        handleFocusChange();
-
-        // Register for change events
-        textField.focusedProperty().addListener((observable, oldValue, newValue) -> {
+        focusListener = (observable, oldValue, newValue) -> {
```

Comment:
we are a bit inconsistent in wrapping (or not) listeners into their weak counterparts - behaviors tend to not :) That's okay - and less crowded by additional code - as long as they are removed properly, IMO. As to the single vs. multiple lines: ersonally, I tend to not change more than absolutely needed - it had the brackets before the fix so I kept them.

Reply:
I'm also interested in the opinion from others. I think we are a bit more safer with weak listener and there are used often as well. 
But as you correctly mentioned a lot of times (still) a listener is created inline. But I think on most occurences where it is not it is wrapped in an weak one.

Reply:
The main idea here is that we must remove the added listeners. 
A Strong listener can cause leak and a WeakListener will not cause leak, but if not removed then a WeakListener would be executed before it gets GCed.
As long as we are removing them properly on disposal, they both look similar to me, but yes we can be consistent in choosing appropriate listener.
IMO, we can keep it out of scope of this PR, It can be taken as a follow up issue and make a consistent change in bulk.

Reply:
okay, leaving as-is - if we'll have too much time in future we can start a overall cleansing round :)